Test workflow failing on Lint step

The test workflow is failing due to some issue with the linter tool.
```
Run golint ./...
/home/runner/work/_temp/25a14027-9883-4832-8711-ab0cdfbcc658.sh: line 1: golint: command not found
Error: Process completed with exit code 127.
```
Looking at the dependencies installation step it seems it did downloaded the package though.
```
go: downloading golang.org/x/lint v0.0.0-20210508222113-6edffad5e616
go: upgraded golang.org/x/lint v0.0.0-20200302205851-738671d3881b => v0.0.0-20210508222113-6edffad5e616
```
Instead of trying to figure out what is going on with the tool I believe it's better to just remove it as the project has been abandoned.
> NOTE: Golint is https://github.com/golang/go/issues/38968. There's no drop-in replacement for it, but tools such as [Staticcheck](https://staticcheck.io/) and go vet should be used instead.

https://github.com/golang/lint

Maybe switch to the mentioned tool [Staticcheck](https://staticcheck.io/).

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -12,17 +12,12 @@ jobs:
 
     steps:
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v4
       with:
         go-version: '1.14'
 
     - name: Checkout
-      uses: actions/checkout@v2
-
-    - name: Install dependencies
-      run: |
-        go version
-        go get -u github.com/axw/gocov/gocov
+      uses: actions/checkout@v4
 
     - name: Code format
       run: test -z "$(gofmt -d . | tee /dev/stderr)"
@@ -30,11 +25,8 @@ jobs:
     - name: Vet
       run: go vet ./...
 
-    - name: Coverage
-      run: go test -race -coverprofile=coverage.txt -covermode=atomic ./...
-
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v1
+    - name: Test
+      run: go test -race -covermode=atomic ./...
 
     - name: Build
       run: go build

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -22,14 +22,10 @@ jobs:
     - name: Install dependencies
       run: |
         go version
-        go get -u golang.org/x/lint/golint
         go get -u github.com/axw/gocov/gocov
 
     - name: Code format
       run: test -z "$(gofmt -d . | tee /dev/stderr)"
-  
-    - name: Lint
-      run: golint ./...
 
     - name: Vet
       run: go vet ./...

